libosrmc/osrmc.cc

```diff
@@ -245,14 +256,40 @@ float osrmc_table_response_duration(osrmc_table_response_t response, unsigned lo
 
   auto& durations = response_typed->values["durations"].get<osrm::json::Array>();
   auto& durations_from_to_all = durations.values.at(from).get<osrm::json::Array>();
-  auto duration = durations_from_to_all.values.at(to).get<osrm::json::Number>().value;
+  auto optional = durations_from_to_all.values.at(to);
+
+  if (optional.is<osrm::json::Null>()) {
+    *error = new osrmc_error{"NoRoute: Impossible route between points"};
```

Comment:
This is the same error that the Route API returns, which makes it easier for us to match on the string in the application. I'm very open to more robust suggestions as well!

Reply:
Returning an error and infinity is fine I think.

But we should document especially the infinity behavior and make it clear to the user that this is sort of a soft error.

The user does not have to shut it down just because there is no route between two sources in a huge table. Instead the user could e.g. use a great circle distance approximation in this case.

Reply:
Great! I'll actually be doing something very similar if I see an `INFINITY`. :-)

I've added a comment to the header file.